### PR TITLE
errtracker: include the build-id of /usr/bin/snap

### PR DESCRIPTION
**Pull request by mvo5**
This will help us figuring out what version of snapd was used on the host even if we re-execed.

### CODE REVIEW

File: errtracker/errtracker.go [hunk 1]
```diff
@@ -66,11 +69,18 @@ func Report(snap, channel, errMsg string) (string, error) {
 
 	crashDbUrl := fmt.Sprintf("%s/%s", CrashDbURLBase, identifier)
 
+	usrBinSnapBuildID, err := osutil.GetBuildID(usrBinSnap)
```
Comment (zyga) [inline errtracker/errtracker.go]: Can you please use dirs.DistroLibExecDir + "snapd" here.

File: errtracker/errtracker_test.go [hunk 2]
```diff
@@ -35,14 +35,15 @@ import (
...
+	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/release"
 )
 
 // Hook up check.v1 into the "go test" runner
 func Test(t *testing.T) { TestingT(t) }
 
 type ErrtrackerTestSuite struct {
-	restorer func()
+	restorer []func()
```
Comment (zyga) [inline errtracker/errtracker_test.go]: nitpick: we have that test case that does this kind of thing automatically, not sure if you want to use it but I wanted to at leat mention it

File: errtracker/errtracker.go (hunk 1 above)
Comment (mvo5) [inline errtracker/errtracker.go]: This is just using /usr/bin/snap -  I could include /usr/lib/snapd/snapd here too, but it seems given that they come from the same package in almost all cases this should give us the data we need(?).

File: errtracker/errtracker_test.go (hunk 2 above)
Comment (mvo5) [inline errtracker/errtracker_test.go]: Oh, which one is that? Happy to rework this one if there is a better pattern :)

File: errtracker/errtracker.go (hunk 1 above)
Comment (zyga) [inline errtracker/errtracker.go]: oh, sorry, I didn't see this  +1

File: errtracker/errtracker_test.go (hunk 2 above)
Comment (zyga) [inline errtracker/errtracker_test.go]: We have `testutil/base.go` with BaseTest that does roughly this

Comment (mvo5) [inline errtracker/errtracker_test.go]: Thanks, added that now.

**Pull request merged**